enhancements/cluster-logging/multi-cluster-log-forwarder.md

```diff
@@ -0,0 +1,217 @@
+
+
+## Motivation
+
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+
+### User Stories
+
+
+* As an administrator of a Red Hat managed cluster, I want to RBAC my log forwarder configuration from customer admins so they can take ownership of their log forwarder needs without being able to modify mine.
+* As an administrator of HyperShift, I want to deploy individual log forwarders to isolate audit log collection of each managed control plane.
+* As an administrator adopting vector, I want to deploy it separately from my existing fluentd deployment so they can operate side-by-side and I can migrate my workloads
+
+
+
+### Goals
+
+* Cluster administrators control which users are allowed to define log collection and which logs they are allowed to collect.
+* Users are able to continue using cluster logging and log forwarding as they have historically in the past
+* Users with allowable permissions are able specify additional log collection configurations
+* Log forwarder deployments are isolated so they do not interfere with other log forwarder deployments
+
+### Non-Goals
+
+* Introduction of the next version of logging APIs.
+
+## Proposal
+
+This is where we get down to the nitty gritty of what the proposal
+actually is. Describe clearly what will be changed, including all of
+the components that need to be modified and how they will be
+different. Include the reason for each choice in the design and
+implementation that is proposed here, and expand on reasons for not
+choosing alternatives in the Alternatives section at the end of the
+document.
+
+This proposal identifies two separate workflows in order to support the legacy deployment and allow additional deployments to meet the enhancement goals.  The legacy deployment will be familiar to users of ClusterLogForwarder prior to the implimentation of this enhancement.  They should see no differences in the manor by which they use log forwarding.  The new workflow will require additional permissions to create new ClusterLogForwarders in order to limit the number of deployments for resource concerns.  Cluster administrators will need to explicitly allow additional deployments
+
+The workflows make the following assumptions:
+
+* The **cluster-logging-operator** is deployed to the *openshift-logging* namespace
+* The **cluster-logging-operator** is able to watch any namespace
+
+### ClusterLogForwarder not NS:openshift-logging/Name:instance
+
+**cluster administrator** is a user:
+
+* responsible for maintaining the cluster
+* able to bind cluster roles to serviceaccounts  
+* that deploys the **cluster-logging-operator**
+
+**namespace administrator** is a user:
+
+* able to create a serviceaccount
```

Comment:
Only a serviceaccount? Not additionally secrets of type serviceaccount-token?

Reply:
@periklis this may be where I lack following changes in OpenShift and service accounts.  Remind me?  OpenShift will no longer be minting SA tokens?  I don't know it matters from the perspective of this enhancement other then needing a token to potentially push to a "default" logstore. We will ask the user to provide the permissions they require for CLF but the operator will need to apply any bindings required to do collection (e.g. mount the node FS for the collector)

Reply:
Yes since 4.12 the k8s core will not create the SA token in a separate secret automatically. You need to create one Secret yourself. As you say it is only needed for potential authentication in-cluster.

Reply:
> Yes since 4.12 the k8s core will not create the SA token in a separate secret automatically. You need to create one Secret yourself. As you say it is only needed for potential authentication in-cluster.

What is it we need to express then here?  Are there usecases to creating an SA without a token?  Are we that usecase?

Reply:
When we use typically ClusterLogForwarder outputs to external systems (i.e. Splunk, Elasticsearch, etc.) we ask users to give us their credentials. In this case we do not need to have a serviceacount token because the systems leave outside the cluster.

However ClusterLogForwarder outputs can point also to system living inside the same cluster w/o our product managing them (i.e. custom Loki, custom ES8). In this case the users might want to the serviceaccount token to authenticate our collector with their storage solution.